src/servant/tt_001/F84.c

```diff
@@ -109,7 +109,104 @@ s32 UpdateEntityVelocityTowardsTarget(s32 unused, s32 targetX, s32 targetY) {
     return D_us_801735C0;
 }
 
-INCLUDE_ASM("servant/tt_001/nonmatchings/F84", func_us_80171284);
+Entity* func_us_80171284(
+    Entity* self) { // Assume self is also an Entity pointer
+    Entity* entity;
+    s32 i;
+    s32 index;
+    u32 matches = 0;
+#if !defined(VERSION_PSP)
+    s32 posDelta;
+#endif
+
+    // Hunt through these entities looking for ones that match all criteria.
+    // Call them a success and increment successes.
+    entity = &D_800762D8[0];
+    for (i = 0; i < 128; i++, entity++) {
+        D_us_801735C4[i] = 0;
+
+        // Very similar code to CheckAllEntitiesValid
+        if (!entity->entityId)
+            continue;
+        if (entity->hitboxState == 0)
+            continue;
+        if (entity->flags & FLAG_UNK_00200000)
+            continue;
+        if (entity->posX.i.hi < -16)
+            continue;
+        if (entity->posX.i.hi > 272)
+            continue;
+        if (entity->posY.i.hi > 240)
+            continue;
+        if (entity->posY.i.hi < 0)
+            continue;
+            // Differs from here
+#if defined(VERSION_PSP)
+        if (entity->hitboxState & 8 &&
+            D_us_80170510[D_us_80173810.level / 10][2] == 0)
+            continue;
+        if (abs(self->posX.i.hi - entity->posX.i.hi) >= 49)
+            continue;
+#else
+        if (entity->hitboxState & 8 &&
+            D_us_80170510[D_us_80173810.level / 10][0] == 0)
+            continue;
+        posDelta = self->posX.i.hi - entity->posX.i.hi;
+        if (posDelta < 0) {
+            posDelta *= -1;
+        }
+        if (posDelta >= 49)
+            continue;
+#endif
+
+#if defined(VERSION_PSP)
+        if (abs(self->posY.i.hi - entity->posY.i.hi) >= 49)
+            continue;
+#else
+        posDelta = self->posY.i.hi - entity->posY.i.hi;
+        if (posDelta < 0) {
+            posDelta *= -1;
+        }
+        if (posDelta >= 49)
+            continue;
+#endif
+
+        if (!self->facingLeft && self->posX.i.hi < entity->posX.i.hi)
+            continue;
+        if (self->facingLeft && self->posX.i.hi > entity->posX.i.hi)
+            continue;
+        if (entity->hitPoints >= 0x7000)
+            continue;
+
+        if (entity->flags & FLAG_UNK_80000) {
+            matches += 1;
+            D_us_801735C4[i] = 1;
+        } else {
+            entity->flags |= FLAG_UNK_80000;
+            return entity;
+        }
+    }
+
+    if (matches != 0) {
+        index = D_us_80173820 % 128;
+
+        for (i = 0; i < 128; i++) {
+            if (D_us_801735C4[index] != 0) {
+#if defined(VERSION_PSP)
+                entity = &D_800762D8[index + 64];
+#else
+                entity = &D_800762D8[index];
```

Comment:
My guess here is that D_800762D8 is a fake symbol and this is index + 64 on both systems